packages/next/src/server/async-storage/request-store.ts

```diff
@@ -234,3 +249,10 @@ function createRequestStoreImpl(
       (globalThis as any).__serverComponentsHmrCache,
   }
 }
+
+export function synchronizeMutableCookies(store: RequestStore) {
+  // TODO: does this need to update headers as well?
```

Comment:
If you're updating the cookies here, you don't need to worry about the headers as they aren't mutable.

Reply:
right, but i think the render that follows the action won't see the `cookie` header in the form that it otherwise would? i need to test that actually

Reply:
to explain it better. the point here is that when we run an action
```
async function action() {
  "use server"
   ;(await cookies()).set("myCookie", "1")
}
```

and we also render the updated version of the page, then that page render should see that cookie value:
```
(await cookies()).get("myCookie").value === '1'
```

but now i'm wondering if the same applies to headers? should the render see `Cookie: myCookie=1` in headers()?